### DISCUSSION TITLE
What is Configuring Global Azure AD Endpoints

### DISCUSSION THREAD

**Discussion by gitbls** [abraunegg/onedrive#2707] [Q&A] [answered]
My backups to onedrive run once per day in the middle of the night. I log the onedrive output and email it to myself.

Since I updated to onedrive v2.5.0-rc1-11-g235fbf7 I've noticed that the step Configuring Global Azure AD Endpoints can take an extraordinarily long time. 

I assume it's due to load on azure or onedrive in the middle of the night, but still seems quite long. and sometimes it's quite fast.
```
** ~13 minutes
2024-04-26 02:15:43 Configuring Global Azure AD Endpoints
2024-04-26 02:28:36 Performing a database consistency and integrity check on locally stored data 
** 6 seconds
2024-04-25 02:15:39 Configuring Global Azure AD Endpoints
2024-04-25 02:15:45 Performing a database consistency and integrity check on locally stored data 
```

**Comment by abraunegg** [comment, marked as answer]
@gitbls 
> What is Configuring Global Azure AD Endpoints

It is merely a log output when you start the application, how the application is configuring itself to use the 'common' Azure AD authentication endpoints.

There is *zero* online interaction at that point in the application execution other than variable assignment.

If this is taking an inordinate amount of time, there is *something* else going on on your system that is being missed in your logging - as if the next line is *truly* `Performing a database consistency and integrity check on locally stored data` ... you are missing a whole bunch of other application output well before it gets to that actual point of execution. This possibly accounts for your query.

Given that you are running `v2.5.0-rc1-11-g235fbf7` - I suggest that you re-compile to get `onedrive v2.5.0-rc1-36-g0f012b9`

Please note also, v2.5.0-rc2 will drop in the next few days.

**Comment by gitbls** [comment]
Thanks for the quick response. I understand that it's log output, and likely an Azure issue, it's just that minutes is...surprising.

And yes, that's what comes from onedrive stdout and stderr (piped to stdout). Here's the onedrive command I'm using. Yes, upload only, master is here. Does that change what would be expected?
```
onedrive --version
onedrive --sync --upload-only --enable-logging 2>&1 | while read line ; do logit "$line" ; done 
logit "ODB Sync Completed"
) | mail -s "$msubj" -r "$mfrom" "$mto"
```

And here's a full log (minus the 3 files copied) from /var/log/onedrive:
```
2024-Apr-23 02:15:38.4991418] Forcing client to use IPv4 connections only
[2024-Apr-23 02:15:39.1962809] Configuring Global Azure AD Endpoints
[2024-Apr-23 02:15:41.4597563] Performing a database consistency and integrity check on locally stored data
[2024-Apr-23 02:23:38.8918839] Changed local items to upload to OneDrive: 3
[2024-Apr-23 02:23:39.6890852] Uploading modified file: 
[2024-Apr-23 02:23:40.0960798] Uploading modified file: 
[2024-Apr-23 02:24:06.0583594] Uploading modified file: 
[2024-Apr-23 02:24:06.7747533] Scanning the local file system '~/OneDrive' for new data to upload
[2024-Apr-23 02:24:25.8018870] New items to upload to OneDrive: 1
[2024-Apr-23 02:24:26.5271897] Uploading new file redacted-file.xxx... done.
[2024-Apr-23 02:24:26.9034440] 
[2024-Apr-23 02:24:26.9034719] Sync with Microsoft OneDrive is complete
```
I'll wait for v2.5.0-rc2 and update then. Thanks for the heads up!

**Reply by abraunegg** [reply]
> `onedrive --sync --upload-only --enable-logging`

As suspected - you are using minimal logging output - thus missing a whole bunch of things that are going on between those 2 log lines - for example:
```
...
Using the following path to store the runtime application log: <redacted>
Using IPv4 and IPv6 (if configured) for all network operations
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  onedrive v2.5.0-rc1-36-g0f012b9
Account Type:         business
Default Drive ID:     <redacted>
Default Root ID:      <redacted>
Remaining Free Space: 765.92 GB (822398716572 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: <redacted>
Performing a database consistency and integrity check on locally stored data
....
```

So - you have 3 potential delay points where Internet access (and any congestion) + your local system CPU processing speed will impact what the application is doing:
1. Post 'Configuring Global Azure AD Endpoints' where the system attempts to initialise the OneDrive API against the OneDrive services
2. Obtaining the 'account details' and 'root' drive details
3. Obtaining the current remaining free space online

Suggest you add `--verbose` to your logging so you know what is going on.

**Comment by gitbls** [comment]
Thanks, added `--verbose` and looking forward to seeing the difference. And, thanks for the responses and insights.

**Comment by abraunegg** [comment]
@gitbls 
FYI v2.5.0-rc2 is now available - please read: https://github.com/abraunegg/onedrive/discussions/2710

A number of memory leaks were identified in rc1 which *should* be resolved in rc2 (testing has verified this).

Please update to v2.5.0-rc2 when possible.

**Reply by gitbls** [reply]
Thanks! Will install tomorrow

**Reply by gitbls** [reply]
Installed new version. Test run works. Now the default. Thanks!

**Reply by abraunegg** [reply]
No more delay in access ?

**Reply by gitbls** [reply]
Not this afternoon. Will report on how it goes with my standard overnight backup.

**Reply by gitbls** [reply]
Regular overnight backup ran fine with no delay in access. Thanks!

**Reply by abraunegg** [reply]
Thanks for the feedback
